generic-helpers/src/main/java/io/elimu/a2d2/helpers/MixPanelService.java

```diff
@@ -0,0 +1,72 @@
+package io.elimu.a2d2.helpers;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.drools.core.spi.KnowledgeHelper;
+
+import io.elimu.a2d2.exception.WorkItemHandlerException;
+import io.elimu.a2d2.genericmodel.ServiceRequest;
+
+public class MixPanelService {
+	private static final MixPanelService INSTANCE = new MixPanelService();
+
+	private ThreadLocal<String> distinctId = ThreadLocal.withInitial(() -> "");
+	
+	private MixPanelService() {
+	}
+
+	public static MixPanelService getInstance() {
+		return INSTANCE;
+	}
+
+	public void identify(String distinctId) {
+		this.distinctId.set(distinctId);
+	}
+	
+	public void identify(ServiceRequest request) {
```

Comment:
I don't think `identify` would ever be used. Clients use this model to identify a single identity (aka `distinctId` - in our case, patient or user that is running an app) and then all following events are associated to that identity. For REST services that process requests for different identities, the `identify` call is not useful.